rosetta/transactor/validator.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package transactor
+
+import (
+	"github.com/onflow/flow-go/model/flow"
+
+	"github.com/optakt/flow-dps/rosetta/identifier"
+)
+
+type Validator interface {
+	Account(rosAccountID identifier.Account) (address flow.Address, err error)
+	Block(rosBlockID identifier.Block) (height uint64, blockID flow.Identifier, err error)
+	Currency(currency identifier.Currency) (symbol string, decimals uint, err error)
```

Comment:
Why do we have named returns here? I'm fine with having named returns in interface definitions but we should be consistent about it then IMO. For now it's not the case everywhere so I'd rather have an issue for standardizing all of our interfaces with named params and returns, and not do it here yet.

Reply:
I personally don't think we should use named returns for interfaces unless they are ambiguous. For example, having `(address flow.Address, err error)` is redundant and doesn't convey any extra information. However, both `(height uint64, blockID flow.Identifier, err error)` and `(symbol string, decimals uint, err error)` makes sense imo, because they disambiguate the returns. So I think that should be the standard.

Reply:
https://github.com/optakt/flow-dps/issues/393